lib/events/athena/querier.go

```diff
@@ -588,3 +592,36 @@ func (ks *keyset) ToKey() string {
 	copy(b[8:24], ks.uid[:])
 	return base64.URLEncoding.EncodeToString(b[:])
 }
+
+var maxUUID = uuid.MustParse("ffffffff-ffff-ffff-ffff-ffffffffffff")
+
+// fromAthenaOrDynamoKeyset parses startKey used for query pagination.
+// It supports also startKey in format used from dynamoevent, to provide
+// smooth migration between dynamo <-> athena backend when event exporter is running.
+func fromAthenaOrDynamoKeyset(startKey string, order types.EventOrder) (*keyset, error) {
+	startKeyset, err := fromKey(startKey)
+	if err != nil {
+		// check if it's dynamoDB staryKey to provide backward compatible.
+		createdAt, inerr := dynamoevents.GetCreatedAtFromStartKey(startKey)
+		if inerr != nil {
+			// can't process it as athena keyset or dynamo, return top level err
+			return nil, trace.Wrap(err)
+		}
+		// createdAt is returned from dynamo startKey, however uid is not stored
+		// there. On athena side for pagination we use following syntax:
+		// (event_time, uid) > (?,?) for ASC order, so let's used 0000 uid there.
+		// In worst case it will resut in few duplicate events.
+		if order == types.EventOrderAscending {
+			return &keyset{
+				t:   createdAt.UTC(),
+				uid: uuid.Nil,
+			}, nil
+		}
+		// For DESC order we use (event_time, uid) < (?,?), so use FFFF as uuid.
+		return &keyset{
+			t:   createdAt.UTC(),
+			uid: maxUUID,
+		}, nil
+	}
+	return startKeyset, nil
+}
```

Comment:
nit: could we refactor this to make it simpler?

1. extract the dynamo key fallback code into a new helper function `fromDynamoKey`
2. rename `fromKey` to `fromAthenaKey` since it only handles Athena keys
3. rename `fromAhtenaOrDynamoKeyset` to `fromKey`
4. make `fromKey` look like this:
```go
// fromKey parses startKey used for query pagination.
// It supports also startKey in format used from dynamoevent, to provide
// smooth migration between dynamo <-> athena backend when event exporter is running.
func fromKey(startKey string, order types.EventOrder) (*keyset, error) {
	startKeyset, athenaErr := fromAthenaKey(startKey)
	if athenaErr == nil {
		return startKeyset, nil
	}
	startKeyset, err := fromDynamoKey(startKey, order)
	if err != nil {
		// can't process it as athena keyset or dynamo, return athena err
		return nil, trace.Wrap(athenaErr)
	}
	return startKeyset, nil
}
```

I pushed a patch commit if you accept these changes, which you can just cherry-pick from: f4a4ea856b3b4ac5005b20d641341ff2414c08bd (in branch: `gavinfrazar/suggested-patch-for-athena-dynamo-migration-pr`)

Reply:
done, thanks!